WalletWasabi/Helpers/Fractional.cs

```diff
@@ -0,0 +1,134 @@
+using NBitcoin;
+
+namespace WalletWasabi.Helpers;
+
+public readonly struct Fraction : IComparable<Fraction>
+{
+	private readonly int _numerator;
+	private readonly int _denominator;
+
+	public Fraction(int numerator, int denominator = 1)
+	{
+		if (denominator == 0)
+		{
+			throw new ArgumentException("Denominator cannot be zero.");
+		}
+
+		var (n, d) = Simplify(numerator, denominator);
+		_numerator = n;
+		_denominator = d;
+	}
+
+	public static implicit operator Fraction(int value) =>
+		new(value);
+
+	public static implicit operator Fraction(FeeRate value) =>
+		value.SatoshiPerByte;
+
+	public static implicit operator Fraction(decimal value)
+	{
+		return CreateFraction(value, 1);
+
+		static Fraction CreateFraction(decimal value, int denominator) =>
+			value % 1 == 0
+				? new Fraction((int)value, denominator)
+				: CreateFraction(value * 10, denominator * 10);
+	}
+
+	public Fraction(decimal numerator, int denominator)
+	{
+		if (denominator == 0)
+		{
+			throw new ArgumentException("Denominator cannot be zero.");
+		}
+
+		int precision = BitConverter.GetBytes(decimal.GetBits(numerator)[3])[2];
+		int scaleFactor = (int)Math.Pow(10, precision);
+		int scaledNumerator = (int)(numerator * scaleFactor);
+		int scaledDenominator = denominator * scaleFactor;
+
+		var (n, d) = Simplify(scaledNumerator, scaledDenominator);
+		_numerator = n;
+		_denominator = d;
+	}
```

Comment:
This contains magic. I would prefer something less performant but much much clear like:

```c#
public Fraction(decimal numerator, int denominator)
{
	var f = (Fraction)numerator / denominator;
	_numerator = f._numerator;
	_denominator = f._denominator;
}
```